Suppression urls non déréférençables

eonm-pro opened the pull request:
Cette PR supprime les URLs qui génèrent des erreurs 404 dans le profil d'application.

Côté SED j'ai fait le choix de chainer deux patterns avec `;` plutôt que d'avoir une seule expression régulière trop complexe.

1.  `s/<a href="(https:\/\/rdafr\.fr\/Elements\/.*?\/)" target="_blank">.*?<\/a>/\1/`
2. `s/<a href="https:\/\/rdafr\.fr\/(Elements|termList)\/.*?>(.*?)<\/a>/\2/`

Dans le premier cas on récupère  et on affiche le contenu de l'attribut href dans la documentation. Dans le second cas on récupère et on affiche la valeur littérale de la balise `<a>` (par exemple pour les termList).

Pour la réécriture du HTML j'ai rajouté une nouvelle entrée RUN dans le Dockerfile pour identifier clairement la "fonctionnalité".

@kerphi J'ai testé la substitution en local à partir du HTML de [test.rda.fr/profil](https://test.rdafr.fr/profil-application/), en revanche je n'ai pas testé la partie Docker.

eonm-pro committed:
Suppression des URLs non déréférençables

eonm-pro committed:
Fix: suppression des URLs non déréférençables

eonm-pro committed:
Merge remote-tracking branch 'origin/suppression-urls-non-dereferencables' into suppression-urls-non-dereferencables

kerphi reviewed (changes requested):
Suggestion de regexp "plus lisible" pour traiter des URL (utilisation de # qui permet d'éviter d'avoir à échapper les / qu'on retrouve souvent dans les URL)

kerphi commented:
Voici la regexp modifiée que je suggère : 
```
sed -i -E 's#<a href="(https://rdafr\.fr/Elements/.*?/)" target="_blank">.*?</a>#\1#;s#<a href="https://rdafr\.fr/(Elemen
ts|termList)/.*?>(.*?)</a>#\2#' /usr/share/nginx/html/profil-application/index.html
```

Autre chose pour gagner en lisibilité : peut-être faire deux sed et ainsi l'éclater sur deux lignes ?

kerphi commented on Dockerfile:
```diff
@@ -41,6 +41,10 @@ RUN curl -F inputShapeFile=@profil-application/rdafr-shacl.ttl \
          > /usr/share/nginx/html/profil-application/index.html
+
+# Suppression des URLs non déréférençables
+RUN sed -i -E 's/<a href="(https:\/\/rdafr\.fr\/Elements\/.*?\/)" target="_blank">.*?<\/a>/\1/;s/<a href="https:\/\/rdafr\.fr\/(Elements|termList)\/.*?>(.*?)<\/a>/\2/' /usr/share/nginx/html/profil-application/index.html
```
En découpant le sed sur deux lignes, ca donnerait ceci ? 
```suggestion
RUN sed -i -E 's#<a href="(https://rdafr\.fr/Elements/.*?/)" target="_blank">.*?</a>#\1#' /usr/share/nginx/html/profil-application/index.html
RUN sed -i -E 's#<a href="https://rdafr\.fr/(Elements|termList)/.*?>(.*?)</a>#\1#' /usr/share/nginx/html/profil-application/index.html
```

eonm-pro replied:
J'avais complètement oublié la possibilité d'utiliser d'autres délimiteurs pour les expressions sed. L'utilisation de `#` rend les choses beaucoup plus lisible 👍.

Pour la scission j'ai opté pour une solution alternative qui nous évite d'avoir à lire le fichier deux fois (même si en terme de performance c'est négligeable).

```
RUN sed -E -i /usr/share/nginx/html/profil-application/index.html \
    -e 's#<a href="(https://rdafr\.fr/Elements/.*?/)" target="_blank">.*?</a>#\1#' \
    -e 's#<a href="https://rdafr\.fr/(Elements|termList)/.*?>(.*?)</a>#\2#'
```

kerphi replied:
Ha excellent le double "-e", je ne connaissais pas.

eonm-pro committed:
Fix: amélioration de la lisibilité de la command SED

Les caractères  délimiteurs  `/` ont été changés par  '# ' pour éviter la multiplication des échappements. 
Les expressions SED ont été scindées.

eonm-pro committed:
Fix: amélioration de la lisibilité de la commande SED

Les caractères  délimiteurs  `/` ont été changés par  '# ' pour éviter la multiplication des échappements.
Les expressions SED ont été scindées.

eonm-pro committed:
Merge remote-tracking branch 'origin/suppression-urls-non-dereferencables' into suppression-urls-non-dereferencables
